acceptance/tests/parser_functions/puppet_lookup_cmd.rb

```diff
@@ -10,8 +10,9 @@
 @coderoot = "#{@testroot}/code"
 @confdir = "#{@testroot}/puppet"
 
-@mastername = on(master, facter('fqdn')).stdout.chomp
-@agentname = on(agent, facter('fqdn')).stdout.chomp
+@agent = agent.is_a?(Array) ? (agent.count > 1 ? agent[1] : agent[0]) : agent  # agent[0] might be on master machine.
```

Comment:
mentioned in meatspace, better to do something like:

```ruby
@agent = [agent].flatten.find { |host| host != master }
```
rather than relying on the ordering of which nodes have the master role.